mapper/cumulocity/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -178,9 +178,17 @@ where
     // so we add a ':'
     let mut date_string: String = Deserialize::deserialize(deserializer)?;
     let str_size = date_string.len();
-    date_string = date_string[0..str_size - 2].to_string()
-        + ":"
-        + &date_string[str_size - 2..str_size].to_string();
+    // check if `date_string` does not have a colon.
+    let date_string_end = &date_string.split('+').last();
+    date_string = match date_string_end {
+        Some(string) if !string.contains(':') => {
+            date_string[0..str_size - 2].to_string()
+                + ":"
+                + &date_string[str_size - 2..str_size].to_string()
+        }
+        _ => date_string,
+    };
+
```

Comment:
Isn't it a good idea to add a unit test to check this?

Reply:
Yes it is.